llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -2515,13 +2550,25 @@ bool SelectionDAG::expandMultipleResultFPLibCall(
     SDValue StoreValue = ST->getValue();
     unsigned ResNo = StoreValue.getResNo();
     Type *StoreType = StoreValue.getValueType().getTypeForEVT(Ctx);
-    if (CallRetResNo == ResNo || !ST->isSimple() ||
+    if (
+        // Ensure the store corresponds to an output pointer.
```

Comment:
Weird formatting, get condition on the same line as the if?

Reply:
This is a (really) long if condition with multiple cases, I added comments to make the checks a little easier to follow.

Reply:
I think it would be better to set some bool variables, put comments above those, and use them in the if statement.

Reply:
I've just gone with multiple ifs (hope thats okay) :slightly_smiling_face:  Using `bool` variables would prevent short circuiting when an earlier case was failed.

Reply:
Ah of course, looks good to me :+1: